pymeasure/instruments/lecroy/lecroyT3DSO1204.py

```diff
@@ -1190,3 +1199,126 @@ def trigger(self):
+                ["C1", "C2", "C3", "C4"], ["C1", "C2", "C3", "C4"]]
+    )
+
+    set_measure_parameter = Instrument.setting(
+        "PACU %s,%s",
+        """ The PARAMETER_CUSTOM command installs a measurement and starts the specified
+        measurement of the specified source.
+        The commands accepts two arguments with the following syntax:
+        measure_parameter = (<parameter>,<source>)
+        <source> := {C1,C2,C3,C4}'
+        <parameter> := one of the following
+        Parameter   Description
+        PKPK        vertical peak-to-peak
+        MAX         maximum vertical value
+        MIN         minimum vertical value
+        AMPL        vertical amplitude
+        TOP         waveform top value
+        BASE        waveform base value
+        CMEAN       average value in the first cycle
+        MEAN        average value
+        RMS         RMS value
+        CRMS        RMS value in the first cycle
+        OVSN        overshoot of a falling edge
+        FPRE        preshoot of a falling edge
+        OVSP        overshoot of a rising edge
+        RPRE        preshoot of a rising edge
+        PER         period
+        FREQ        frequency
+        PWID        positive pulse width
+        NWID        negative pulse width
+        RISE        rise-time
+        FALL        fall-time
+        WID         Burst width
+        DUTY        positive duty cycle
+        NDUTY       negative duty cycle
+        ALL         All measurement """,
+        validator=_set_measure_parameter_validator,
+        values=[_measurable_parameters, ["C1", "C2", "C3", "C4"]]
+    )
+
+    def get_measure_parameter(self, parameter: str, source: str):
+        """ The PARAMETER_VALUE query measures and returns the specified measurement value
+        present on the selected waveform.
+        The commands accepts two arguments with the following syntax:
+        get_measure_parameter = (<source>, <parameter>) be careful because this is the opposite
+        order as the set_measure_parameter property
+        <source> := {C1,C2,C3,C4}
+        <parameter> := same as the set_measure_parameter property
+        """
+
+        parameter, source = _set_measure_parameter_validator(value=(parameter, source),
+                                                             values=[self._measurable_parameters,
+                                                                     ["C1", "C2", "C3", "C4"]])
+        output = self.ask("%s:PAVA? %s" % (source, parameter))
+        match = re.match(r'^\s*(?P<parameter>\w+),\s*(?P<value>.*)\s*$', output)
+        if match:
+            if match.group('parameter') != parameter:
+                raise ValueError(f"Parameter name {match.group('parameter')} does not match with "
+                                 f"expected one {parameter}")
+            return float(match.group('value'))
+        else:
+            raise ValueError(f"Cannot extract value from output {output}")
```

Comment:
These are strange property names. Normally you use a property (either python properties or our property creators, `control` etc.) instead of getter/setter methods.

I see, that a property does not work for the getter, as you have to tell the source. Here `get...`is fine. But `set_...` as the name of a property does not sound right.

These properties would be more appropriate for the channel object, such that you do not have to choose the channel manually.

Reply:
I agree with all your points. Actually, at first, I tried to implement these properties as a single channel property as you suggest.
The reason why I split this into two properties (one to set the measurement and one to get it), is that these are two different SCPI commands with quite different syntaxes. I think this is a design flaw in the oscilloscope firmware but there is nothing that I can do about it.
The reason why I detached it from the channel is that, from the user's point of view, it is easier to automatize all measurements if they are all in the same place. For example, the `measure_delay` command refers to multiple channels at the same time. Otherwise, the client should look for measurable quantities scattered in many places.
In conclusion, the reason for the current implementation is the ease of use from the client side.

Reply:
I see your points.
I'd make both functions a (plain python) property of the channel (just named `measure_parameter`), as it is a channel property. Then you can make two methods in the main class (i.e. here) which get/set the channel property.
Would that be fine with you?

`set_measure_parameter` should be a method, not a property (based on the name).

Reply:
In commit 5bbabe5a6b9bb38bd61639dc6883f7ffcf9cfe5f I did something very close to what you are suggesting. Using a plain python property would not work because there is the additional `parameter` argument. I have also changed the method names to make them less ambiguous.